src/common/datatypes/Map.h

```diff
@@ -28,7 +29,23 @@ struct Map {
   explicit Map(const std::string& json) {
     auto obj = folly::parseJson(json);
     for (auto& kv : obj.items()) {
-      kvs.emplace(kv.first.getString(), Value(kv.second));
+      // for now, only limited primitive types in value field are supported
+      //   no nested type(list, map, set) is supported
+      if (kv.second.isString()) {
+        kvs.emplace(kv.first.asString(), Value(kv.second.asString()));
+      } else if (kv.second.isInt()) {
+        kvs.emplace(kv.first.asString(), Value(kv.second.asInt()));
+      } else if (kv.second.isDouble()) {
+        kvs.emplace(kv.first.asString(), Value(kv.second.asDouble()));
+      } else if (kv.second.isBool()) {
+        kvs.emplace(kv.first.asString(), Value(kv.second.asBool()));
+      } else if (kv.second.isNull()) {
+        kvs.emplace(kv.first.asString(), Value());
+      } else {
+        LOG(ERROR) << "Only string, int64, double, bool, null are supported in "
+                      "parseJson to construct Map, now trying to parse from: "
+                   << kv.second.typeName();
+      }
     }
```

Comment:
It's better to move the implementation to .cpp file.

Reply:
> The ctest failed, please fix it.

working on it now,

> It's better to move the implementation to .cpp file.

will do so.

Reply:
done :)